packages/rearch/lib/rearch.dart

```diff
@@ -225,6 +228,28 @@ class CapsuleContainer implements Disposable {
       manager.dispose();
     }
   }
+
+  /// Warms up the specified capsules by waiting for them to all be [AsyncData].
+  Future<void> warmUp<T extends AsyncValue<dynamic>>(
+    Iterable<Capsule<T>> capsules,
+  ) async {
+    final completer = Completer<()>();
+    final dispose = listen((use) {
+      final resolved = capsules.map((capsule) => use(capsule));
+
+      if (resolved.every((capsule) => capsule is AsyncData)) {
+        completer.complete(());
+      }
+
+      final asyncError = resolved.whereType<AsyncError<dynamic>>().firstOrNull;
+      if (asyncError != null) {
+        completer.completeError(asyncError.error);
```

Comment:
can you pass in the asyncError.stackTrace into completeError too?

Reply:
til